packages/orn-locator/src/resolvers/books.ts

```diff
@@ -358,93 +348,41 @@ export const element = async({bookArchiveVersion, bookId, bookContentVersion, pa
   };
 };
 
-export const elementSearch = async(query: string, limit: number, filters: {[key: string]: string | string[]} = {}): Promise<Awaited<ReturnType<typeof element>>[]> => {
-  const scopes = 'scope' in filters
-    ? (await locateAll(typeof filters.scope === 'string' ? [filters.scope] : filters.scope))
-      .filter(isResourceOrContentOfTypeFilter(['book']))
-      .map(book => book.id)
-    : (await library('all')).contents.map(book => book.id);
-
-  const books = await Promise.all(scopes.map(async bookId => {
-    const {archivePath, bookVersion} = await getArchiveInfo(bookId);
-    const archiveVersion = archivePath.replace('/apps/archive/', '');
-    return `${archiveVersion}/${bookId}@${bookVersion}`;
-  }));
+export const elementSearch = async(searchClient: SearchClient, query: string, limit: number): Promise<Awaited<ReturnType<typeof element>>[]> => {
+  const releaseIds = await getReleaseIds();
+  const results = await doOpenSearch(searchClient, limit, query, releaseIds, 'i1');
 
-  return fetch(`https://openstax.org/open-search/api/v0/search?q=${query}&books=${encodeURIComponent(books.join(','))}&index_strategy=i1&search_strategy=s1`)
-    .then(response => response.json())
-    .then(json => json.hits.hits.slice(0, limit) as any[])
-    .then(results => Promise.all(results.map(result => {
-      const bookId = result._index.match(/__(.*)@/)[1];
-      const pageId = result._source.page_id.split('@')[0];
-      const elementId = result._source.element_id;
-      return element({bookId, pageId, elementId});
-    })));
+  return Promise.all(results.map(result => {
+    const bookId = assertInstanceOf<string[]>(result.index.match(/__(.*)@/), Array)[1];
+    const pageId = result.source.pageId.split('@')[0];
+    const elementId = result.source.elementId;
+    return element({bookId, pageId, elementId});
```

Comment:
this will re-resolve the data, i had wanted to store the entire orn object in the search index so that when we get here we can just return it without re-resolving it, can we do that? ( it seems like the other two functions do that already?)

Reply:
The other 2 functions were the ones I added... I have to check what's stored in i1 to see if it has the bookId

Reply:
bookId is missing, but I can probably split the element function to avoid calling pageWithData.

Reply:
OK there are a few things missing from the data stored in i1 in open-search that we'd have to add or make a new index... basically everything in pageWithData is returned, but the following are missing in i1:

- Book slug, page slug (to return and to build the Rex URL)
- page title (contextTitle)
- bookData from CMS

Reply:
oh is this element search using the pre-existing element search used by rex?... that explains the difference.... 

i'm just worried that if we leave this like this it'll slow down the search results for the other resource types. if anything we'd want to just remove this resource from the default search

Reply:
Done, excluded element search from the default search